archive/100-theorems-list/45_partition.lean

```diff
@@ -0,0 +1,615 @@
+import ring_theory.power_series
+import combinatorics.composition
+import data.nat.parity
+import data.finset.nat_antidiagonal
+import tactic.interval_cases
+import tactic.apply_fun
+
+open power_series
+noncomputable theory
+
+variables {α : Type*}
+
+lemma eq_mul_inv_iff [field α] {a b c : power_series α} (h : constant_coeff _ c ≠ 0) :
+  a = b * c⁻¹ ↔ a * c = b :=
+⟨λ k, by simp [k, mul_assoc, power_series.inv_mul _ h],
+ λ k, by simp [← k, mul_assoc, power_series.mul_inv _ h]⟩
+
+lemma eq_inv_iff [field α] {a b : power_series α} (h : constant_coeff _ b ≠ 0) : a = b⁻¹ ↔ a * b = 1 :=
+by rw [← eq_mul_inv_iff h, one_mul]
+lemma power_series.inv_eq_iff [field α] {a b : power_series α} (h : constant_coeff _ b ≠ 0) : b⁻¹ = a ↔ a * b = 1 :=
+by rw [eq_comm, eq_inv_iff h]
+
+open finset
+open_locale big_operators
+
+lemma count_repeat_ite {α : Type*} [decidable_eq α] (a b : α) (n : ℕ)  :
+  multiset.count a (multiset.repeat b n) = if (a = b) then n else 0 :=
+begin
+  split_ifs,
+    cases h,
+    rw multiset.count_repeat,
+  apply multiset.count_eq_zero_of_not_mem,
+  intro,
+  apply h,
+  apply multiset.eq_of_mem_repeat a_1,
+end
+
+open_locale classical
+open power_series
+
+noncomputable theory
+
+/-- A composition of `n` is a list of positive integers summing to `n`. -/
+@[ext] structure partition (n : ℕ) :=
```

Comment:
See also `combinatorics/composition.lean`

Reply:
Ahh - I copied the definition from there and forgot to change this bit; thanks! I think I'll put some of this stuff in `combinatorics/partition.lean` and make a PR for that.